app/authorization.go

```diff
@@ -102,6 +102,18 @@ func (a *App) SessionHasPermissionToUser(session model.Session, userId string) b
 	return false
 }
 
+func (a *App) SessionHasPermissionToUserOrBot(session model.Session, userId string) bool {
+	if a.SessionHasPermissionToUser(session, userId) {
+		return true
+	}
+
+	if err := a.SessionHasPermissionToManageBot(session, userId); err == nil {
```

Comment:
Where is this method defined? I don't see it anywhere, so perhaps that's why your build is failing?

Reply:
Oh, I didn't realize this was to merge into the `bot` branch, not master. Never mind then

Reply:
`SessionHasPermissionToManageBot` is currently only in the `bots` branch to which this PR is being merged.

Unfortunately, the current test failures are unrelated -- just non-deterministic unit tests again. I've got a PR to address at least some of them here: https://github.com/mattermost/mattermost-server/pull/10117.